### PR TITLE
add org.kde.kget

### PR DESCRIPTION
**Pull request by Amit0617**
- ### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Amit0617** [comment]
bot, build org.kde.kget

**Comment by Amit0617** [comment]
bot, build org.kde.kget

File: org.kde.kget.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
+{
+  "id" : "org.kde.kget",
+  "branch" : "master",
```
Comment (hfiguiere) [inline org.kde.kget.json:3]:
```suggestion
```

File: org.kde.kget.json [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  "runtime" : "org.kde.Platform",
+  "runtime-version" : "5.15-21.08",
+  "runtime-commit" : "e8e3a76d0c26375c55740919779a70e76843cca2305ec01bc3c5d3cbeae2c9b6",
```
Comment (hfiguiere) [inline org.kde.kget.json:6]:
```suggestion
```

File: org.kde.kget.json [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  "sdk" : "org.kde.Sdk",
+  "sdk-commit" : "5a0b113ebd0a557ee77d73107176ff5f9489da8db422a41f03e2159d43b078a8",
```
Comment (hfiguiere) [inline org.kde.kget.json:8]:
```suggestion
```

File: org.kde.kget.json [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  "command" : "kget",
+  "modules" : [
+    {
+      "name" : "boost",
+      "buildsystem" : "simple",
+      "sources" : [
+        {
+          "url" : "https://boostorg.jfrog.io/artifactory/main/release/1.76.0/source/boost_1_76_0.tar.bz2",
+          "sha256" : "f0397ba6e982c4450f27bf32a2a83292aba035b827a5623a14636ea583318c41",
+          "type" : "archive"
```
Comment (hfiguiere) [inline org.kde.kget.json:18]:
```suggestion
          "type" : "archive"
          "url" : "https://boostorg.jfrog.io/artifactory/main/release/1.76.0/source/boost_1_76_0.tar.bz2",
          "sha256" : "f0397ba6e982c4450f27bf32a2a83292aba035b827a5623a14636ea583318c41"
```

same below. it is common practice that "type" is first when declaring a source.

File: org.kde.kget.json [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    "--socket=x11",
+    "--socket=wayland",
+    "--filesystem=host",
+    "--share=network"
+  ],
+  "tags" : [
+    "nightly"
+  ],
+  "rename-icon" : "kget",
+  "desktop-file-name-suffix" : " (Nightly)"
```
Comment (hfiguiere) [inline org.kde.kget.json:139]: all of these are usually toward the top of the manifest.

```suggestion
  "finish-args" : [
    "--share=ipc",
    "--socket=fallback-x11",
    "--socket=wayland",
    "--filesystem=host",
    "--share=network"
  ],
  "rename-icon" : "kget"
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.kde.kget.json [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+          "type" : "archive"
+        }
+      ],
+      "build-commands" : [
+        "./bootstrap.sh --prefix=/app --with-libraries=system",
+        "./b2 -j $FLATPAK_BUILDER_N_JOBS install"
+      ]
+    },
+    {
+      "name" : "gpgme",
```
Comment (hfiguiere) [inline org.kde.kget.json:26]: gpgme is in the runtime.

File: org.kde.kget.json [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+          "type" : "archive"
+        }
+      ],
+      "cleanup" : [
+        "/lib/libgpgme*.la",
+        "/lib/libqgpgme.la"
+      ],
+      "modules" : [
+        {
+          "name" : "libgpg-error",
```
Comment (hfiguiere) [inline org.kde.kget.json:40]: libgpg-error is in the runtime

File: org.kde.kget.json [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+              "sha256" : "fc07e70f6c615f8c4f590a8e37a9b8dd2e2ca1e9408f8e60459c67452b925e23",
+              "type" : "archive"
+            }
+          ],
+          "cleanup" : [
+            "/lib/libgpg-error.la"
+          ],
+          "modules" : [
+            {
+              "name" : "gettext",
```
Comment (hfiguiere) [inline org.kde.kget.json:53]: gettext is in the runtime.

File: org.kde.kget.json [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+                  "url" : "https://ftp.gnu.org/pub/gnu/gettext/gettext-0.19.8.tar.gz",
+                  "sha256" : "3da4f6bd79685648ecf46dab51d66fcdddc156f41ed07e580a696a38ac61d48f",
+                  "type" : "archive"
+                }
+              ]
+            }
+          ]
+        },
+        {
+          "name" : "libassuan",
```
Comment (hfiguiere) [inline org.kde.kget.json:65]: this is in the runtime too.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.kde.kget.json (hunk 6 above)
Comment (Amit0617) [reply org.kde.kget.json:33]: OK, how can i know this? So will be able to keep in mind for next time.

Comment (Amit0617) [reply org.kde.kget.json:33]: is this the logic that those which doesn't have buildsystems are in runtime

Comment (Amit0617) [reply org.kde.kget.json:33]: if yes then confirm that these shouldn't be included.

**Comment by hfiguiere** [comment]
bot, build org.kde.kget

File: org.kde.kget.json [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+    "--share=network"
+  ],
+  
+  "modules" : [
+    {
+      "name" : "boost",
+      "buildsystem" : "simple",
+      "sources" : [
+        {
+          "type" : "archive"
```
Comment (hfiguiere) [inline org.kde.kget.json:22]:
```suggestion
          "type" : "archive",
```

File: org.kde.kget.json [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+  "runtime" : "org.kde.Platform",
+  "runtime-version" : "5.15-21.08",
+  "sdk" : "org.kde.Sdk",
+  "command" : "kget",
+  "rename-icon" : "kget",
+  "finish-args" : [
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--filesystem=host",
```
Comment (hfiguiere) [inline org.kde.kget.json:12]: I am of the opinion this should be limited to `xdg-download`

Comment (Amit0617) [reply org.kde.kget.json:12]: i will need little bit explanation too

**Comment by Amit0617** [comment]
bot, build org.kde.kget

**Comment by nedrichards** [comment]
@flathub/kde FYI

File: org.kde.kget.json [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+    {
+      "name" : "kdelibs4support",
+      "buildsystem" : "cmake-ninja",
+      "config-opts" : [
+        "-DENABLE_TESTING=OFF"
+      ],
+      "sources" : [
+        {
+          "url" : "https://invent.kde.org/frameworks/kdelibs4support.git",
+          "commit" : "65eedb15cc81391f72a7dbebcb6861c46a58fb45",
```
Comment (tsdgeos) [inline org.kde.kget.json:95]: tarball not git commit

File: org.kde.kget.json [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+          "type" : "git"
+        }
+      ]
+    },
+    {
+      "name" : "libktorrent",
+      "buildsystem" : "cmake-ninja",
+      "sources" : [
+        {
+          "url" : "https://invent.kde.org/kde/libktorrent.git",
```
Comment (tsdgeos) [inline org.kde.kget.json:116]: tarball not git commit

File: org.kde.kget.json [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+          "type" : "git"
+        }
+      ]
+    },
+    {
+      "name" : "kget",
+      "buildsystem" : "cmake-ninja",
+      "sources" : [
+        {
+          "url" : "https://invent.kde.org/kde/kget.git",
```
Comment (tsdgeos) [inline org.kde.kget.json:127]: tarball not git commit

File: org.kde.kget.json (hunk 6 above)
Comment (travier) [reply org.kde.kget.json:33]: https://github.com/flathub/org.kde.kontact/pull/55#issuecomment-1000528230 I need to check this again but the FDO Runtime provided gpgme is missing Qt support.

File: org.kde.kget.json (hunk 11 above)
Comment (travier) [reply org.kde.kget.json:12]: The idea is that KGet does not need to access your entire home directory. It just needs a place to download files and xdg-download is just that (see the Freedesktop spec).

Comment (Amit0617) [reply org.kde.kget.json:12]: understood. thanks

**Comment by Amit0617** [comment]
bot, build org.kde.kget

File: org.kde.kget.json [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+  ],
+  
+  "modules" : [
+    {
+      "name" : "boost",
+      "buildsystem" : "simple",
+      "sources" : [
+        {
+          "type" : "archive",
+          "url" : "https://boostorg.jfrog.io/artifactory/main/release/1.76.0/source/boost_1_76_0.tar.bz2",
```
Comment (travier) [inline org.kde.kget.json:23]: This is an "old" version. I think the current one is 1.78.

File: org.kde.kget.json (hunk 8 above)
Comment (travier) [reply org.kde.kget.json:60]: This is not resolved

File: org.kde.kget.json (hunk 7 above)
Comment (travier) [reply org.kde.kget.json:47]: This is indeed not needed AFAIK. See https://github.com/flathub/org.kde.kontact/pull/55/commits/07642fa2a444bdcb68cf5d1a54b038f55de7b243

File: org.kde.kget.json (hunk 9 above)
Comment (travier) [reply org.kde.kget.json:72]: Same as above. Not resolved

**Review by travier — CHANGES_REQUESTED** [review]

File: org.kde.kget.json (hunk 15 above)
Comment (Amit0617) [reply org.kde.kget.json:23]: do i need to update link for those too which are already in runtime? like gpgme etc.

File: org.kde.kget.json (hunk 8 above)
Comment (Amit0617) [reply org.kde.kget.json:60]: do i have to remove them completely? i have asked above from him but didn't got reply

File: org.kde.kget.json (hunk 7 above)
Comment (Amit0617) [reply org.kde.kget.json]: thanks for reference.

File: org.kde.kget.json (hunk 8 above)
Comment (travier) [reply org.kde.kget.json]: You should try building without them first. If you get an error then we will have to see if this is really needed.

File: org.kde.kget.json [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+        }
+      ]
+    },
+    {
+      "name" : "libktorrent",
+      "buildsystem" : "cmake-ninja",
+      "sources" : [
+        {
+          "type" : "archive",
+          "url" : "https://download.kde.org/stable/ktorrent/5.2.0/libktorrent-2.2.0.tar.xz",
```
Comment (tsdgeos) [inline org.kde.kget.json:77]: this needs to be libktorrent 21.12.1 from the release service

**Comment by Amit0617** [comment]
bot, build org.kde.kget.json

**Comment by Amit0617** [comment]
bot, build org.kde.kget

**Comment by Amit0617** [comment]
bot, build org.kde.kget

**Comment by travier** [comment]
LGTM

**Comment by tsdgeos** [comment]
There is a problem when closing the app with the [x]

The app doesn't "really" close because it should go to Plasma's notification area, but it doesn't do that (the one installed from the distro does).

Can you try to check what's wrong?

**Comment by travier** [comment]
Good catch, we probably need something like what's done in ktorrent.

**Comment by Amit0617** [comment]
I hope that was required change

**Comment by Amit0617** [comment]
bot, build org.kde.kget

**Comment by travier** [comment]
LGTM. We now have proper system tray support.

**Comment by tsdgeos** [comment]
Looks reasonable to me too :+1:

**Pull request closed without merging**